### PR TITLE
LSTESData fixes

### PR DESCRIPTION
**Pull request by ariostas**
I started working on the LSTESData-related comments. In particular, these comments:
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1674648911
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1674789755
https://github.com/cms-sw/cmssw/pull/45117#discussion_r1674793908

I removed the alpaka functions from places where they are not needed. Matti said that it would be better two have two ES products for host-only and device-dependent data, so I'll have to figure that out.

### CODE REVIEW

**Comment by ariostas** [comment]
It compiles, but I'm not sure if it actually works, so I'll check.

/run standalone

File: RecoTracker/LSTCore/src/alpaka/Event.h [hunk 1]
```diff
@@ -87,9 +87,9 @@ namespace lst {
     const uint16_t nLowerModules_;
     const unsigned int nPixels_;
     const unsigned int nEndCapMap_;
-    const std::shared_ptr<const ModulesBuffer<Device>> modulesBuffers_;
-    const std::shared_ptr<const PixelMap> pixelMapping_;
-    const std::shared_ptr<const EndcapGeometryBuffer<Device>> endcapGeometryBuffers_;
+    ModulesBuffer<Device> const& modulesBuffers_;
+    std::shared_ptr<const PixelMap> pixelMapping_;
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/Event.h:91]: why not a  `PixelMap const& pixelMapping_;` here as well?

Comment (ariostas) [reply RecoTracker/LSTCore/src/alpaka/Event.h:91]: Yeah, I'll make it more consistent

**Comment by ariostas** [comment]
I'm a bit confused why the plots look different, but I'll look into it.

**Comment by ariostas** [comment]
@slava77 let's merge batch4 and I'll figure out where the differences are coming from next week

**Comment by ariostas** [comment]
Should be good now.

/run all

**Review by slava77 — APPROVED** [review]

**Pull request merged**